### PR TITLE
Use 4-Byte UTF-8 Unicode Encoding

### PR DESCRIPTION
**Pull request by jacquesbh**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

### CODE REVIEW

**Review by pamil — COMMENTED** [review]
What are the implications of this change for existing databases? Does it change the behaviour for columns length?

**Comment by jacquesbh** [comment]
@pamil For me, no impact on column length or indices after 5.7.7. According to my searches.

**Comment by pamil** [comment]
Indeed, looks like a change that we should do then! `utf8` is an alias for `utf8mb3` which is deprecated in favour of `utf8mb4`. I'm just wondering how to upgrade existing projects to `uft8mb4`, considering these concerns here: https://serversforhackers.com/c/mysql-utf8-and-indexing

**Comment by pamil** [comment]
Indeed, looks like a change that we should do then! `utf8` is an alias for `utf8mb3` which is deprecated in favour of `utf8mb4`. I'm just wondering how to upgrade existing projects to `uft8mb4`, considering these concerns here: https://serversforhackers.com/c/mysql-utf8-and-indexing

**Comment by jacquesbh** [comment]
The issue withe the indices is gone after 5.7.7 :).

File: config/packages/doctrine.yaml [hunk 1]
```diff
@@ -9,6 +9,6 @@ doctrine:
     dbal:
         driver: 'pdo_mysql'
         server_version: '5.7'
-        charset: UTF8
+        charset: UTF8mb4
```
Comment (pamil) [inline config/packages/doctrine.yaml:12]:
```suggestion
        charset: utf8mb4
```

That looks like an official spelling, according to MySQL documentation.

**Review by pamil — COMMENTED** [review]
Should we generate some migrations for this change to take place for existing data?

**Comment by jacquesbh** [comment]
> Should we generate some migrations for this change to take place for existing data?

I don't think so… maybe a warning in the UPGRADE?

**Review by pamil — APPROVED** [review]

**Comment by pamil** [comment]
Thanks, Jacques! :tada:

**Pull request merged**